clusterloader2/pkg/measurement/common/generic_query_measurement.go

```diff
@@ -182,7 +183,8 @@ func (g *genericQueryGatherer) String() string {
 
 func (g *genericQueryGatherer) query(q GenericQuery, executor QueryExecutor, startTime, endTime time.Time) ([]*model.Sample, error) {
 	duration := endTime.Sub(startTime)
-	boundedQuery := fmt.Sprintf(q.Query, measurementutil.ToPrometheusTime(duration))
+	// Replace all provided duration placeholders (%v) with the test duration.
+	boundedQuery := strings.ReplaceAll(q.Query, "%v", fmt.Sprintf("%v", measurementutil.ToPrometheusTime(duration)))
```

Comment:
I think it could be  `measurementutil.ToPrometheusTime(duration)` instead of `fmt.Sprintf("%v", measurementutil.ToPrometheusTime(duration))`?

Reply:
You are right. The `ToPrometheusTime` returns a string. Thanks for point it out.

I made the change.